File: languages/js/test/rolesHelpers.ts
export class Org {
  readonly name: string;

  constructor(name: string) {
    this.name = name;
  }
}

export class Repo {
  readonly name: string;
  readonly org: Org;

  constructor(name: string, org: Org) {
    this.name = name;
    this.org = org;
  }
}

export class Issue {
  readonly name: string;
  readonly repo: Repo;

  constructor(name: string, repo: Repo) {
    this.name = name;
    this.repo = repo;
  }
}

export class Role {
  readonly name: string;
  readonly resource: Org | Repo;

  constructor(name: string, resource: Org | Repo) {
    this.name = name;
    this.resource = resource;
  }
}

export class User {
  readonly name: string;
  readonly roles: Role[];

  constructor(name: string, roles: Role[]) {
    this.name = name;
    this.roles = roles;
  }
}

File: languages/ruby/spec/oso/polar/roles_helpers.rb
# frozen_string_literal: true

module RolesHelpers
  class Org
    attr_reader :name

    def initialize(name)
      @name = name
    end
  end

  class Repo
    attr_reader :name, :org

    def initialize(name, org)
      @name = name
      @org = org
    end
  end

  class Issue
    attr_reader :name, :repo

    def initialize(name, repo)
      @name = name
      @repo = repo
    end
  end

  class Role
    attr_reader :name, :resource

    def initialize(name, resource)
      @name = name
      @resource = resource
    end
  end

  class User
    attr_reader :name, :roles

    def initialize(name, roles)
      @name = name
      @roles = roles
    end
  end
end
